### PR TITLE
StreamRequestBody shouldn't read more data than actual need.

### PR DESCRIPTION
**Pull request by newacorn**
The StreamRequestBody feature on the server side should not read content that does not belong to the current request body.This is more logical and consistent with the result of not using the StreamRequestBody feature.
Fixes: [#1816](https://github.com/valyala/fasthttp/issues/1816). 
The reason for addressing this issue is that the Read method of requestStream assumes it should only read up to the Content-Length number of bytes.

### CODE REVIEW

**Comment by newacorn** [comment]
https://github.com/valyala/fasthttp/blob/1fb34531652380677219b7c8c9c3d02da007a80b/http.go#L2286
Here, contentLength is guaranteed to be greater than or equal to 0 because the ContinueReadBodyStream function has already filtered out cases where contentLength is less than 0. It is assumed that a negative Content-Length value other than -1 should not be processed：
https://github.com/valyala/fasthttp/blob/1fb34531652380677219b7c8c9c3d02da007a80b/http.go#L1372

Below code will also handle cases where maxBodySize is less than contentLength.
https://github.com/valyala/fasthttp/blob/1fb34531652380677219b7c8c9c3d02da007a80b/http.go#L2295

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]
Thanks, the fix looks good. But the new file needs to be formatted. And http.go needs some more newlines:
```
  http.go:2287: line is 147 characters (lll)
  		A fixed-length pre-read function should be used here; otherwise, it may read content beyond the request body into areas outside of the br buffer.
  http.go:2288: line is 160 characters (lll)
  	        This could affect the handling of the next request in the br buffer, if there is one. The original two branches can be handled with this single branch.
```

File: http.go [hunk 1]
```diff
@@ -2283,11 +2283,13 @@ func readBodyWithStreaming(r *bufio.Reader, contentLength, maxBodySize int, dst
...
-	} else {
-		b, err = readBodyIdentity(r, readN, dst)
-	}
+	// A fixed-length pre-read function should be used here; otherwise,
+	// it may read content beyond the request body into areas outside
+	// the br buffer. This could affect the handling of the next request
+	// in the br buffer, if there is one. The original two branches can
+	// be handled with this single branch. by the way,
+	// fix issue: https://github.com/valyala/fasthttp/issues/1816
+	b, err = appendBodyFixedSize(r, dst, readN)
```
Comment (erikdubbelboer) [inline http.go:2293]:
```suggestion
```
Sorry, our strict linter doesn't like this empty line here.

File: server_test.go [hunk 2]
```diff
@@ -4358,3 +4358,51 @@ func (cl *testLogger) Printf(format string, args ...any) {
...
+		err = resp.Read(bufio.NewReader(cliCon))
+		if err != nil {
+			t.Error(err)
+		}
+		err = cliCon.Close()
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+	//
```
Comment (erikdubbelboer) [inline server_test.go:4390]:
```suggestion

```

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]

**Comment by newacorn** [comment]
@erikdubbelboer  May I ask what tools you use to lint the fasthttp codebase?

**Comment by erikdubbelboer** [comment]
@newacorn you can see that here: https://github.com/valyala/fasthttp/blob/38a91cda6fa7e351834c1a7d97db9b89feec78c2/.github/workflows/lint.yml#L26-L28

**Comment by erikdubbelboer** [comment]
Thanks!

**Pull request merged**